fuel-tests/tests/tx.rs

```diff
@@ -122,6 +122,47 @@ async fn submit() {
     assert_eq!(tx.id(), ret_tx.id());
 }
 
+#[tokio::test]
+async fn dependent_tx() {
+    let srv = FuelService::new_node(Config::local_node()).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    let gas_price = 0;
+    let gas_limit = 1_000_000;
+    let maturity = 0;
+    let byte_price = 0;
+
+    let script = vec![
+        Opcode::ADDI(0x10, REG_ZERO, 0xca),
+        Opcode::ADDI(0x11, REG_ZERO, 0xba),
+        Opcode::LOG(0x10, 0x11, REG_ZERO, REG_ZERO),
+        Opcode::RET(REG_ONE),
+    ];
+    let script: Vec<u8> = script
+        .iter()
+        .flat_map(|op| u32::from(*op).to_be_bytes())
+        .collect();
+
+    let tx = fuel_tx::Transaction::script(
+        gas_price,
+        gas_limit,
+        byte_price,
+        maturity,
+        script,
+        vec![],
+        vec![],
+        vec![],
+        vec![],
+    );
+
+    let id = client.submit(&tx).await.unwrap();
+
+    let dependent_txs = client.dependent_transactions(&id.to_string()).await;
+
+    println!("{:?}", dependent_txs);
+    assert!(dependent_txs.unwrap().is_some());
```

Comment:
The test will only guarantee this api returns _something_ in the future. For example, if someone changed the server-side API to just return `Transaction::Default` the test would tell them everything was still ok.

One way to help with coming up with better test assertions is to think through all the criteria and behaviors a user might expect from this API, and then verify those are achieved. At least from a happy path perspective for integration tests like these. We can assume negative edge cases should be mostly handled at the unit test level.

One way to write these down might look like this:
* as a user of the find_dependent txs API, I expect that the transaction id I'm searching from is included in the response.
* as a user of the find_dependent txs API, I expect that transactions with overlapping txo inputs and outputs will form an ordered priority list based on their dependencies and then by gas price.

Then the test should verify scenarios written out like this, ensuring continuing functionality in the face of future changes and refactors.